class-udm-updater.php

```diff
@@ -999,5 +921,119 @@ protected function disconnect() {
+					if ((isset($decoded['data']) && isset($decoded['data']['plugin_info']) && isset($decoded['data']['plugin_info']['x-spm-duplicate-of'])) || (isset($option['duplicate_site_is_connected']) && $option['duplicate_site_is_connected'])) {
+						$option['duplicate_site_is_connected'] = true;
+						$decoded['duplicate_site_is_connected'] = true;
+						$this->update_option($this->option_name, $option);
+						ob_start();
+						$this->disconnect();
+						$connection_result = ob_get_clean();
+						$decoded = json_decode($connection_result, true);
+						if (!is_array($decoded)) $decoded = array();
+						if (isset($decoded['code']) && 'OK' === $decoded['code']) {
+							delete_site_option('udmanager_updater_sid');
+							$option['duplicate_site_is_connected'] = false;
+							$option['was_previously_sharing_licence'] = true;
+							$this->update_option($this->option_name, $option);
+							$this->connect();
+							exit;
+						} else {
+							unset($option['email']); // though the disconnect() method will unset this, it'd be good if we also unset this just in case the disconnect() method fails and returns an unrecognised code
+						}
+					} else {
+						if (isset($option['was_previously_sharing_licence'])) {
+							$decoded['was_previously_sharing_licence'] = $option['was_previously_sharing_licence'];
+							unset($option['was_previously_sharing_licence']);
+						}
+						if (isset($option['duplicate_site_is_connected'])) unset($option['duplicate_site_is_connected']);
+					}
+					$this->update_option($this->option_name, $option);
+					$result['body'] = json_encode($decoded);
+				}
+
+				echo $result['body'];
+				
+				if (isset($decoded['data']) && isset($decoded['data']['plugin_info'])) {
+					$plugin_info = $decoded['data']['plugin_info'];
+					
+					$this->get_puc_updates_checker();
+					
+					// e.g. Puc_v4p6_Plugin_UpdateChecker
+					$checker_class = get_class($this->plug_updatechecker);
+					
+					// Hopefully take off the 'Checker'. The setUpdate() call below wants a compatible version.
+					$plugin_update_class = substr($checker_class, 0, strlen($checker_class)-7);
+					
+					if (class_exists($plugin_update_class) && is_callable(array($plugin_update_class, 'fromObject')) && !empty($this->plug_updatechecker)) {
+
+						// $plugin_update_class::fromObject() is invalid syntax on PHP 5.2
+						$plugin_update = call_user_func(array($plugin_update_class, 'fromObject'), $plugin_info);
+						
+						$update_checker = $this->plug_updatechecker;
+						
+						$installed_version = $update_checker->getInstalledVersion();
+
+						if (null !== $installed_version) {
+							$state = $update_checker->getUpdateState();
+							$state->setLastCheckToNow()->setCheckedVersion($installed_version);
+							$state->setUpdate($plugin_update);
+							$state->save();
+						}
+						
+					}
+
+				}
+				
+			}
+		} elseif (is_wp_error($result)) {
+			echo json_encode(array(
```

Comment:
@andtown 

> "As a result of the implementation, the connect block of code needs to be moved into its own method, which is what this PR does."

It looks like you've also introduced some changes, in these lines here which create output?

Reply:
@DavidAnderson684 Yes, I should have explained this earlier in my previous comment, the reason why I changed the output is that to make it uniform with the `Disconnect` method, also instead of just echoing the output, I think it'd be good if the error message could be shown directly to the user in the dialog box.

Reply:
@andtown But if the output format there is changed, doesn't there need to also be another change to the code that parses that output?

Reply:
@DavidAnderson684 Yes, you're right, there's another PR that handles that already, here is the PR: https://github.com/DavidAnderson684/simba-plugin-manager-updater/pull/11. Well actually the output is still in string format, the only difference is that the old format is in a plain text whereas the new one is a JSON formatted string. So, if you've ever wondered whether or not merging this would break the master branch, I don't think it would.